src/Symfony/Component/HttpKernel/DataCollector/DataCollector.php

```diff
@@ -100,4 +105,24 @@ protected function varToString($var)
 
         return $this->valueExporter->exportValue($var);
     }
+
+    /**
+     * @return callable[] The casters to add to the cloner
+     */
+    protected function getCasters()
+    {
+        return array(
+            '*' => function ($v, array $a, Stub $s, $isNested) {
+                if (!$v instanceof Stub) {
+                    foreach ($a as &$v) {
```

Comment:
@nicolas-grekas have you checked whether the usage of a reference here could be the reason why https://github.com/symfony/symfony/issues/23110 happens (replacing the EntityManager somewhere it should not be replaced) ?